docs/syntax-attributes-functions-parameters.md

```diff
@@ -0,0 +1,61 @@
+# Function Attribute Parameters
+
+## Summary
+
+This RFC proposes a syntax for function attribute parameters. This is a follow up to the [Attributes](./syntax-attributes-functions.md) RFC, which proposed a design for function attributes. We only propose a syntactic extension in this RFC; the rest of the implementation details remain unchanged.
+
+## Motivation
+
+The [Attributes](./syntax-attributes-functions.md) RFC provides a syntax for parameterless function attributes. This suffices for some potential use cases, such as `@native`, and `@inline`. However, other potential cases such as `@deprecated` will benefit from an expressive attribute syntax that allows parameters to be supplied. We can use the parameters to provide the name of the function to be used in place of the deprecated function, and the reason for deprecation. This information can be used to generate informative deprecation warning messages. Another potential use case could be an `@unroll` attribute for loop unrolling which could use a numeric parameter for specifying the number of iterations to be unrolled. This would also require supporting attributes on loops, which would be a topic of discussion for a separate RFC. It might be desirable to allow these attributes to assume a default value for parameters not provided explicitly.
+
+## Design
+
+The following syntax is proposed for attributes:
+
+```ebnf
+parameter-table = '{' '}'
+          | '{' parameter (sep parameter)* '}'
+
+sep = ',' | ';'
+
+parameter = literal 
+          | NAME '=' literal
+
+literal = BOOLEAN | NUMBER | STRING | NIL | parameter-table
+
+list-attribute = NAME parameter-table
+               | NAME
+
+attribute = '@' NAME 
+          | '@[' list-attribute (',' list-attribute)* ']'
+```
+
+In Luau scripts, attributes are specified before the `function` keyword in function definitions. In declaration files, attributes are specified in function type declarations, before the `function` keyword in `declare function (type, ...) : type` syntax, and before the `(` in the `name: (type, ...) : type` syntax. The primary extension proposed to the [Attributes](./syntax-attributes-functions.md) RFC is a new delimited syntax `@[]` for specifying multiple comma-separated attributes with parameters supplied through an optional Luau table. The important features are:
+
+1. Inside the attribute list, `@[]`, attribute names are not allowed to have a leading `@`.
+2. Attribute lists cannot be nested. Attributes cannot be specified on attribute parameters.
+3. Attributes inside `@[]` are separated by a `,`.
+4. Attributes inside `@[]` are allowed to take an optional table of parameters.
+5. The "parameter table" is a Luau table whose entries specify attribute parameters, with or without names, separated by commas or semicolons.
```

Comment:
In table arguments, entries should always have a name. If you wanted no names, use `()`.

Reply:
`()` for attribute parameters is not supported. You can have entries without names in the parameter table. Parameter table is just like a normal Luau table with restrictions on entry value types.